apps/web/src/components/Search/Community/CommunitySearchResult.tsx

```diff
@@ -1,21 +1,36 @@
+import { useRouter } from 'next/router';
 import { Route } from 'nextjs-routes';
-import { useMemo } from 'react';
+import { useCallback, useMemo } from 'react';
 import { graphql, useFragment } from 'react-relay';
 
 import CommunityProfilePicture from '~/components/ProfilePicture/CommunityProfilePicture';
 import { CommunitySearchResultFragment$key } from '~/generated/CommunitySearchResultFragment.graphql';
+import { MentionType } from '~/shared/hooks/useMentionableMessage';
 import { LowercaseChain } from '~/shared/utils/chains';
 
 import SearchResult from '../SearchResult';
+import { SearchResultVariant } from '../SearchResults';
 
 type Props = {
+  keyword: string;
   communityRef: CommunitySearchResultFragment$key;
+  variant: SearchResultVariant;
+
+  onSelect?: (item: MentionType) => void;
+  onClose?: () => void;
 };
 
-export default function CommunitySearchResult({ communityRef }: Props) {
+export default function CommunitySearchResult({
+  communityRef,
+  keyword,
+  variant,
+  onSelect,
+  onClose,
```

Comment:
I think there's opportunity to simplify/consolidate the `onSelect` and `onClose` props - looking at the usage, it looks like they're both meant to be callbacks for when the user selects a search result: onSelect is for the mention flow, and onClose is for the regular search flow. so I think we can make that distinction clearer 

also to me the name "onClose" is a bit confusing because it implies a callback that's called when something closes, but it looks like it's passing in a function that triggers the search view to close. it's another onSelect callback.

ideally we can pass in a single onSelect callback, which can conditionally do different things based on whether its a regular search or mentions search. I know you added a `isMentionSearch` prop in the mobile app, maybe can use a similar thing here?

Reply:
Thats a good call! I've clean up this flow to make it more consistent. Plus its easier to add the tracking